executor/aggfuncs/func_stddevpop.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"math"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/set"
+)
+
+type baseStdDevPopAggFunc struct {
+	baseAggFunc
+}
+
+type stdDevPop4Float64 struct {
+	baseStdDevPopAggFunc
+}
+
+type partialResult4StdDevPopFloat64 struct {
+	count    int64
+	sum      float64
+	variance float64
+}
+
+func (e *stdDevPop4Float64) AllocPartialResult() (pr PartialResult, memDelta int64) {
+	return PartialResult(&partialResult4StdDevPopFloat64{}), 0
+}
+
+func (e *stdDevPop4Float64) ResetPartialResult(pr PartialResult) {
+	p := (*partialResult4StdDevPopFloat64)(pr)
+	p.count = 0
+	p.sum = 0
+	p.variance = 0
+}
+
+func (e *stdDevPop4Float64) AppendFinalResult2Chunk(sctx sessionctx.Context, pr PartialResult, chk *chunk.Chunk) error {
+	p := (*partialResult4StdDevPopFloat64)(pr)
+	if p.count == 0 {
+		chk.AppendNull(e.ordinal)
+		return nil
+	}
+	varicance := p.variance / float64(p.count)
+	chk.AppendFloat64(e.ordinal, math.Sqrt(varicance))
+	return nil
+}
+
+//func calculateIntermediate(count int64, sum float64, input float64, variance float64) float64 {
+//	t := float64(count)*input - sum
+//	variance += (t * t) / (float64(count * (count - 1)))
+//	return variance
+//}
+
+func (e *stdDevPop4Float64) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (memDelta int64, err error) {
```

Comment:
You do not need to implement this function, just let `stdDevPop4Float64` inherit from `varPop4Float64`, and override the `AppendFinalResult2Chunk` method.